### PR TITLE
Add a readme to the theories folder and reorder some of them

### PR DESCRIPTION
**Pull request by scarlehoff**
I've added a small readme with the agreement we reached in Amsterdam a few months ago and modified some theories. I'm still expecting a few iterations and changes before we get to a truly stable theory numbering scheme but I think we are getting there!

The changes I've done are:

```bash
# Baselines NNLO / NLO
20000000 -> 41100000
20000001 -> 41100010

# Scale varied NNLO
20001001 -> 41100001
20001002 -> 41100002
20001003 -> 41100003
20001004 -> 41100004
20001005 -> 41100005
20001006 -> 41100006
20001007 -> 41100007
20001008 -> 41100008
# Scale variations only for jets/dijets
20001009 -> 41100054
20001010 -> 41100055

# Scale varied NLO
20002001 -> 41100011
20002002 -> 41100012
20002003 -> 41100013
20002004 -> 41100014
20002005 -> 41100015
20002006 -> 41100016
20002007 -> 41100017
20002008 -> 41100018
```

<details>
<summary>script</summary>

```bash
#!/usr/bin/env bash

prefix=""
suffix=".yaml"
cp_command="git mv"

change_th_number() {
    name_orig=${prefix}${1}${suffix}
    name_targ=${prefix}${2}${suffix}

    ${cp_command} ${name_orig} ${name_targ} 2>/dev/null

    sed_target=${name_targ}
    sed -i '' "s/${1}/${2}/" ${sed_target}
    # Now try with `_`
    under_1=$(echo ${1} | rev | sed 's/\(...\)/\1_/g' | rev)
    under_2=$(echo ${2} | rev | sed 's/\(...\)/\1_/g' | rev)
    sed -i '' "s/${under_1}/${under_2}/" ${sed_target}
}

# Baselines NNLO / NLO
change_th_number 20000000 41100000
change_th_number 20000001 41100010

# Scale varied NNLO
change_th_number 20001001 41100001
change_th_number 20001002 41100002
change_th_number 20001003 41100003
change_th_number 20001004 41100004
change_th_number 20001005 41100005
change_th_number 20001006 41100006
change_th_number 20001007 41100007
change_th_number 20001008 41100008
# Scale variations only for jets/dijets
change_th_number 20001009 41100054
change_th_number 20001010 41100055

# Scale varied NLO
change_th_number 20002001 41100011
change_th_number 20002002 41100012
change_th_number 20002003 41100013
change_th_number 20002004 41100014
change_th_number 20002005 41100015
change_th_number 20002006 41100016
change_th_number 20002007 41100017
change_th_number 20002008 41100018
```

</details>

### CODE REVIEW

**Comment by Radonirinaunimi** [comment]
Thanks a lot @scarlehoff for this! I only have one small comment: given that we are also still using polarized PCH, shouldn't we use `41_101_ZZZ`?

**Comment by scarlehoff** [comment]
If you think it makes more sense I can make that change, but note that `1XY` is not "all the polarized", it is just me thinking in binary and flipping 0s and 1s

**Comment by Radonirinaunimi** [comment]
> If you think it makes more sense I can make that change, but note that `1XY` is not "all the polarized", it is just me thinking in binary and flipping 0s and 1s

Ah, ok. If the `_1YY_` are not only polarized theories then that's fine.

**Review by Radonirinaunimi — APPROVED** [review]

File: nnpdf_data/nnpdf_data/theory_cards/readme.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- `_000_`: default 4.0 theories, nFONLL with intrinsic charm.
+- `_001_`: 4.0 theories, nFONLL with perturbative charm.
+- `_002_`: 4.0 theories, nFONLL, N3LO
+
+### 41
+- `41_000_000`: NNPDF4.1 baseline
+- `_100_`: Theories for NNPDFpol2.0
+
+## Old theories
+Theories below < 1000 correspond to fits for the NNPDF4.0 global analysis (and before)
```
Comment (giacomomagni) [inline nnpdf_data/nnpdf_data/theory_cards/readme.md:29]:
```suggestion
Theories below <= 1026 correspond to fits for the NNPDF4.0 global analysis (and before)
```

**Review by giacomomagni — APPROVED** [review]

**Pull request merged**